compiler/parser/RdrHsSyn.hs

```diff
@@ -3219,37 +3266,22 @@ mkInterfaceDecl tycon decls = do
                  pure $ mkTemplateDataDecl l ifChoiceName info
             | L l (InterfaceChoice InterfaceChoiceSignature{..} _) <- decls
             ]
-    pure (toOL (cls : existential : existentialInstance : existentialExerciseInstances ++ ifaceInstances ++ choiceInstances ++ choiceTys ++ choiceDecls))
+    pure (toOL (existential : existentialImplements ++ existentialExerciseInstances ++ ifaceMethods ++ ifaceInstances ++ choiceInstances ++ choiceTys ++ choiceDecls))
```

Comment:
we should really stop calling this existentials (yes I started this …). Let’s leave it out of this PR though

Reply:
Well it kind of is, right? I guess in plain Haskell (well, plain + GADTs) we'd do something like

```
data Token where
  MkToken :: forall t. Implements t Token => t -> Token
```

Reply:
yeah but there aren’t any existential pattern matches or something so I think the name is rather confusing.

Reply:
yeah, fair